### PR TITLE
Add confirm password validation to reset view

### LINKED ISSUE #15909: Password can be reset wiithout new password confirmation.
#### Issue Description

User can reset their password by only fiilling out new password field without filling out password confirmation.

How to reproduce it:

1. Go to [https://www.freecodecamp.org/forgot](https://www.freecodecamp.org/forgot) and try to reset your password.
2. Open a confirmation email from FCC to be directed to reset password page.
3. Fill out your new password on the **New Password field** and **leave the Confirm Password blank.**
4. Hit Change Password button.
5. Try to login with your new password.

#### Browser Information

* Browser Name, Version: Google Chrome Version 58.0.3029.110 (64-bit)
* Operating System: Linux Ubuntu 16.04
* On Desktop

#### Screenshot

screenshot from 2017-09-23 01-35-43
screenshot from 2017-09-23 01-36-09
screenshot from 2017-09-23 01-36-34
screenshot from 2017-09-23 01-37-07
screenshot from 2017-09-23 01-37-23
screenshot from 2017-09-23 01-37-51
screenshot from 2017-09-23 01-38-10

### PR DESCRIPTION
**Pull request by shovanch**
#### Pre-Submission Checklist

- [ ] Your pull request targets the `staging` branch of freeCodeCamp.
- [X] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [X] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [X] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [X] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [X] Tested changes locally.
- [X] Closes currently open issue (replace XXXX with an issue no): Closes #15909 

#### Description

Added confirm password validation to reset view. Shows error message if passwords don't match. Blocks form submission until passwords are validated

### CODE REVIEW

File: server/views/account/reset.jade [hunk 1]
```diff
@@ -7,11 +7,25 @@ block content
...
       .form-group
         label(for='confirm') Confirm Password
-        input.form-control(type='password', name='confirm', value='', placeholder='Confirm password')
+        input.form-control(type='password', id='confirm-password', required, name='confirm', value='', placeholder='Confirm password')
       .form-group
         button.btn.btn-primary.btn-reset(type='submit')
           i.fa.fa-keyboard-o
           | Change Password
+
+        script.
```
Comment (raisedadead) [inline server/views/account/reset.jade]: Move this line and everything below it (the whole script block), to its correct indentation. Your script block is going to be within the form elements without that. Most browsers will not execute such code.

You should have it rather at the same level as line 4 in this snippet.

This is how the jade(pug) templating engine renders elements, and hence indentation is very crucial.

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Comment by raisedadead** [comment]
Are you able to test this locally?

**Comment by shovanch** [comment]
Yes, I have tested them locally, Seems to work fine.

**Comment by shovanch** [comment]
I have amended the requested changes

**Comment by raisedadead** [comment]
> Yes, I have tested them locally.

Do you mind providing some screenshots, I would be able to QA it soon, but a quick look at how it looks would help in doing it faster.

**Comment by shovanch** [comment]
Sure. Should I rather make a GIF showing the process?

**Comment by shovanch** [comment]
I have made a GIF showing the whole process - [GIF link](https://i.imgur.com/LWY8DFB.gif)

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
#### :sparkles: :sparkles: **LGTM.** :sparkles: :sparkles: 

Thanks @shovanch , and congratulations on your first contribution to freeCodeCamp.

/cc @BerkeleyTrue

**Pull request merged**